addons/website_event_track/static/src/interactions/website_event_track.js

```diff
@@ -1,62 +1,58 @@
-publicWidget.registry.websiteEventTrack = publicWidget.Widget.extend({
-    selector: '.o_wevent_event',
-    events: {
-        'input #event_track_search': '_onEventTrackSearchInput',
-    },
+        this.searchText = "";
+        this.agendaScroller = this.el.querySelector(".o_we_agenda_horizontal_scroller_container");
+        this.agendaScrollerElement = this.agendaScroller?.querySelector(".o_we_agenda_horizontal_scroller");
+        this.agendas = Array.from(this.el.querySelectorAll(".o_we_online_agenda"));
+        this.tracks = Array.from(this.el.querySelectorAll(".event_track"));
 
-    //--------------------------------------------------------------------------
-    // Handlers
-    //--------------------------------------------------------------------------
+        if (this.agendas.length > 0) {
+            this.checkAgendasOverflow(this.agendas);
+        }
 
-    /**
-     * @override
-     */
-    start: function () {
-        this._super.apply(this, arguments).then(() => {
-            this.$el.find('[data-bs-toggle="popover"]').popover();
-
-            this.agendas = Array.from(this.target.getElementsByClassName('o_we_online_agenda'));
-
-            if (this.agendas.length > 0) {
-                this._checkAgendasOverflow(this.agendas);
-                this.agendaScroller = this.$el[0].querySelector('.o_we_agenda_horizontal_scroller_container');
-                this.agendaScrollerElement = this.agendaScroller.querySelector('.o_we_agenda_horizontal_scroller');
-
-                this.agendas.forEach(agenda => {
-                    agenda.addEventListener('scroll', event => {
-                        this._onAgendaScroll(agenda, event);
-                        if(this.agendaScroller && this.visibleAgenda) {
-                            this.agendaScroller.scrollLeft = this.visibleAgenda.scrollLeft;
-                        }
-                    });
-                });
-
-                if (this.agendaScroller) {
-                    this._updateAgendaScroll = debounce(this._updateAgendaScroll, 50);
-                    if (document.querySelector('#wrapwrap.event')) {
-                        window.addEventListener('scroll',
-                            this._updateAgendaScroll.bind(this)
-                        );
-                    }
-
-                    window.addEventListener('resize', () => {
-                        this._updateAgendaScroll();
-                    });
-
-                    this.agendaScroller.addEventListener('scroll', () => {
-                        if (this.visibleAgenda) {
-                            this.visibleAgenda.scrollLeft = this.agendaScroller.scrollLeft;
-                        }
-                    });
-
-                    this._updateAgendaScroll();
-                }
-            }
-        });
-    },
+        if (this.agendaScroller) {
+            this.updateAgendaScroll = this.debounced(this.updateAgendaScroll, 50);
+            this.updateAgendaScroll();
+        }
```

Comment:
Does this work ? Didn't `dynamicContent` keep track of the non-debounced function and keep calling it ?
See if:
```js
this.updateAgendaScroll = () => {
    console.log("used");
    this.updateAgendaScroll()
};
```
is actually called by the `t-on-*`.
If not, then in the `t-on-*`: `"t-on-scroll": () => this.updateAgendaScroll(),`